tools/install-deps.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+# This script is used for CI testing.
+
+# Get all dependencies needed for the docker environment
+# This works because perl and YAML::PP are already installed in the container.
+
+getdeps() {
+    perl -MYAML::PP=Load -wE'
+    my $d = Load(do { local $/; <> });
+    for (@{ $d->{targets}->{docker} }) {
+        push @d, grep {!/^%/} keys %{ $d->{$_ . "_requires"} }
+    };
+    say for sort @d' < dependencies.yaml
+}
+
+DEPS=($(getdeps))
+
+#echo "DEPS: ${DEPS[@]}"
+
+sudo zypper --no-refresh install -y -C ${DEPS[@]}
```

Comment:
I consider the approach I suggested easier than the additional shell script with embedded procedural perl code (At least the double for-loop looks quite unusual. In that case I would go for a `say sort map { … } grep { … } …`)

Reply:
The problem is, I think your solution could break:
https://github.com/os-autoinst/os-autoinst/pull/1423#issuecomment-639269454
if we ever add
```yaml
something else with:
  a key: and a value
```
Then we need to change the command anyway.

Reply:
This is the shortest code (with some whitespace left) I came up with:
```
perl -MYAML::PP=Load -0wE'my $d = Load(<>);
say for sort grep {!/^%/} map { keys %{$d->{$_."_requires"}} }
@{$d->{targets}->{docker}}' < dependencies.yaml
```

Reply:
I like that more :) The positive point about your solution is that it uses a proper YAML parser, the negative is still the "perl embedded in shell script, called by container script" but it's ok

Reply:
I rearranged it a bit now.
There is `tools/tools.sh` which simply defines the function `getdeps_docker`.
It can be sourced from the script that installs dependencies.

We could also use it in Docker, do we don't have to generate the Dockerfile.
We only need to extract `dependencies.yaml` and `tools/tools.sh` in OBS. If we ever need to change/add something, we can simply edit `tools.sh`.